nimPNG.nim

```diff
@@ -1728,19 +1731,20 @@ proc convert*(png: PNG, colorType: PNGcolorType, bitDepth: int): PNGResult =
   let modeOut = newColorMode(colorType, bitDepth)
   let size = getRawSize(header.width, header.height, modeOut)
   let numPixels = header.width * header.height
-  let input = cstring(png.pixels)
 
   new(result)
   result.width  = header.width
   result.height = header.height
   result.data   = newString(size)
-  var output = cstring(result.data)
 
   if modeOut == modeIn:
-    copyMem(output, input, size)
```

Comment:
Unfortunately it's nearly impossible to do semantically equivalent copyMem and zeroMem in JS. Even if it is, it will have to recursively traverse the data structure inside the array (like for example deep copying does), which totally breaks expectations of `copyMem` and `zeroMem` performance.

Endian swap - already done: https://github.com/jangko/nimPNG/pull/4/files#diff-f5191b897b8126e1f666e431554a26c9R334